src/Hazelcast.Net/DistributedObjects/ReliableTopicExceptionEventHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Threading.Tasks;
+using Hazelcast.Models;
+
+namespace Hazelcast.DistributedObjects
+{
+    /// <summary>
+    /// Represents a handler for the <see cref="TopicEventTypes.Exception"/> event.
+    /// </summary>
+    /// <typeparam name="T">The reliable topic object type.</typeparam>
+    internal class ReliableTopicExceptionEventHandler<T> : IReliableTopicEventHandler<T>
+    {
+        private readonly Func<IHReliableTopic<T>, ReliableTopicExceptionEventArgs, ValueTask> _handler;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ReliableTopicExceptionEventHandler{T}"/> class.
+        /// </summary>
+        /// <param name="handler">An action to execute</param>
+        public ReliableTopicExceptionEventHandler(Action<IHReliableTopic<T>, ReliableTopicExceptionEventArgs> handler)
+        {
+            _handler = (sender, args) =>
+            {
+                handler(sender, args);
+                return default;
+            };
+        }
+
+        /// <summary>
+        /// Gets event arguments of the event. It is useful to check <see cref="ReliableTopicExceptionEventArgs.Cancel"/>.
+        /// </summary>
+        public ReliableTopicExceptionEventArgs EventArgs { get; private set; }
```

Comment:
There can be only one `Exception` handler (no method chaining). I terminated the fluent method calls at `Exception` since it changes the `args.Cancel`.

If we create the `ExceptionArgs` and pass to `HandleAsync` then we have to change the `IReliableTopicEventHandler` with an overload which accepts an `EventArgs` but each type event has different type of argument. So, each event becomes different type because `IReliableTopicEventHandler` should be generic to type of arguments. In this case, the polymorphism in here disappears. Or, all event arguments share same base class. Also, `EventArgs` property is internal. Do you still want me to change the implementation?

Reply:
OK, I see what you have done here. But, the fact that there is no method chaining does not prevent me from doing:
```
await reliableTopic.SubscribeAsync(on =>
{
  on.Exception((sender, args) => DoSomething(sender, args));
  on.Exception((sender, args) => DoSomethingElse(sender, args));
});
```
Thus registering *two* handlers for the event. So I would think... we should align the `Exception` event with the others, just let ppl chain methods and register as many handlers as they want. Each handler will get the same event args, each handler can flip `Cancel`, what's important is the value of `Cancel` after *all* handlers have executed. That is how other .NET events that support cancellation work.

Then I realize... there are events *handlers* (plural) and event *handler* (singular). I believe `ReliableTopicEventHandler<T>` should be `ReliableTopicEventHandlers<T>` - see how it works for `IMap`. Helps make things less confusing.

Next... see maps, `ReliableTopicEventHandlers<T>` inherits `EventHandlersBase<IReliableTopicEventHandlerBase>` see it's `IReliableTopicEventHandlerBase` here not `IReliableTopicEventHandler` so ... each particular event handler can have it's own different event args = you can have one for exceptions. And create the args outside the handler, and run the handler that can have many methods, on the same args instance.

Reply:
I see your concern. So, I've separated all event (message, exception and terminated) into different type of handlers and event arguments. I think current state https://github.com/hazelcast/hazelcast-csharp-client/pull/825/commits/af597343e5f35c8f65133dcdf2fd911e363588b6 allays your concerns.